apps/studio/components/layouts/SQLEditorLayout/SQLEditorMenu.tsx

```diff
@@ -140,10 +139,25 @@ export const SQLEditorMenu = () => {
               />
             </DropdownMenuTrigger>
             <DropdownMenuContent align="end" side="bottom" className="w-48">
-              <DropdownMenuItem className="gap-x-2" onClick={() => handleNewQuery()}>
-                <FilePlus size={14} />
+              <DropdownMenuItemTooltip
+                className="gap-x-2"
+                disabled={isCheckingPermissions || !canCreateSQLSnippet}
+                tooltip={{
+                  content: {
+                    text: canCreateSQLSnippet
+                      ? undefined
+                      : 'Your do not have sufficient permissions to create new queries',
+                  },
+                }}
+                onClick={() => handleNewQuery()}
+              >
+                {isCheckingPermissions ? (
+                  <Loader2 className="animate-spin" strokeWidth={1.5} size={16} />
+                ) : (
+                  <FilePlus size={14} />
+                )}
```

Comment:
@joshenlim lmk if this loading state works for you. I didn't have an example on hand for menu items, but felt like this would be fine to show a spinner icon here